docs/embedding/full-app-embedding.md

```diff
@@ -4,161 +4,227 @@ redirect_from:
+```
+MAX_SESSION_AGE=2040
+```
 
-E.g., the URL
+To automatically clear a person's login cookies when they end a browser session:
 
 ```
-http://localhost:3000/?top_nav=false&side_nav=false
+MB_SESSION_COOKIES=true
 ```
 
-would disable the top and side nav:
+To log everyone out of Metabase, load the following URL (for example, in a hidden iframe on the logout page of your application):
 
-![Top nav and side nav disabled](./images/no-top-no-side.png)
+```
+https://metabase.yourcompany.com/auth/logout
+```
 
-### `top_nav`
+If you're using [JWT](../people-and-groups/authenticating-with-jwt.md) for SSO, we recommend setting the `exp` (expiration time) property to a short duration (e.g., 1 minute).
 
-Entire top navigation bar, with optional search and new button.
+## Filling an entire iframe with an embedded Metabase page
 
-![Top nav bar](./images/top-nav.png)
+To make an embedded Metabase page fill up the entire iframe (e.g., a question page), use [postMessage()](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) to send a "frame" message from Metabase:
 
-The top bar is hidden by default. Additionally, if you enable the top bar (`top_nav=true`), you can turn on other hidden-by-default top bar options:
+```
+{ “metabase”: { “type”: “frame”, “frame”: { “mode”: “normal” }}}
+```
 
-- `search`: Search bar within the top nav.
-- `new_button`: “New” CTA that lets users create questions and more.
+## Fitting an iframe to a Metabase page with a fixed size
 
-### `side_nav`
+To specify the size of an iframe so that it matches an embedded Metabase page (e.g., a dashboard page), use [postMessage()](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) to send a "frame" message from Metabase:
 
-The main navigation bar.
+```
+{ “metabase”: { “type”: “frame”, “frame”: { “mode”: “fit”, height: HEIGHT_IN_PIXELS }}}
+```
 
-![Side nav](./images/side-nav.png)
+## Keeping an embedding URL up to date
 
-The navigation sidebar is hidden by default, aside from `/collection` and home page product routes. If you want people to be able to minimize the sidebar, you MUST enable the `top_nav`.
+To get an updated embedding URL (e.g., for deep linking), use [postMessage()](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) to send a "location" message _from_ your embedded Metabase:
 
-### `header`
+```
+{ “metabase”: { “type”: “location”, “location”: LOCATION_OBJECT }}
+```
+
+And, to send a "location" message _to_ your embedded Metabase:
 
-The header, visible by default, only applies to dashboards and questions. The header refers to the info and buttons above a question or dashboard. The header includes the title, additional info, and the action buttons (Filter and Summarize).
+```
+{ “metabase”: { “type”: “location”, “location”: LOCATION_OBJECT_OR_URL }}
+```
 
-If you enable the header, these sub-components are visible by default:
+## Showing or hiding Metabase UI components
```

Comment:
@nllho to mention double encoding here as well, re: https://metabase.zendesk.com/agent/tickets/12156